xrpl/faucet/testnet.go

```diff
@@ -0,0 +1,55 @@
+package faucet
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+const (
+	TESTNET_FAUCET_HOST = "faucet.altnet.rippletest.net"
+	TESTNET_FAUCET_PATH = "/accounts"
+)
+
+var _ FaucetProvider = (*TestnetFaucetProvider)(nil)
+
+// TestnetFaucetProvider implements the FaucetProvider interface for the XRPL Testnet.
+// It provides functionality to interact with the Testnet faucet for funding wallets.
+type TestnetFaucetProvider struct {
+	host        string
+	accountPath string
+}
+
+// NewTestnetFaucetProvider creates and returns a new instance of TestnetFaucetProvider
+// with predefined Testnet faucet host and account path.
+func NewTestnetFaucetProvider() *TestnetFaucetProvider {
+	return &TestnetFaucetProvider{
+		host:        TESTNET_FAUCET_HOST,
+		accountPath: TESTNET_FAUCET_PATH,
+	}
+}
+
+// FundWallet sends a request to the Testnet faucet to fund the specified wallet address.
+// It returns an error if the funding request fails.
+func (fp *TestnetFaucetProvider) FundWallet(address string) error {
```

Comment:
The faucet also support an optional `Amount` field in the body to define the amount to send. By default it is 100 if it is not provided. It would be nice to add it as well (and for devnet too).